### PR TITLE
[Feature](nereids)support view and nested view

### PR DESCRIPTION
**Pull request by sohardforaname**
# Proposed changes

Issue Number: close #xxx

## Problem summary

support view in query
and add a rewrite rule: merge consecutive projects.
the rule can merge relative consecutive projects to one project to improve efficiency

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [-] No
    - [ ] I don't know
2. Has unit tests been added:
    - [-] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [-] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [-] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [-] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/EliminateAliasNode.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+/**
+ * Eliminate the logical sub query and alias node after analyze and before rewrite
+ * If we match the alias node and return its child node, in the execute() of the job
+ * <p>
+ * TODO: refactor group merge strategy to support the feature above
+ */
+public class EliminateAliasNode implements AnalysisRuleFactory {
+    @Override
+    public List<Rule> buildRules() {
+        return ImmutableList.of(
```
Comment (wangshuo128) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/EliminateAliasNode.java:36]: Why not use ```logicalSubQueryAlias().then(alias -> alias.child()) ```for all the cases?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Alias.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ public class Alias extends NamedExpression implements UnaryExpression {
 
     private final ExprId exprId;
     private final String name;
-    private final List<String> qualifier;
+    private List<String> qualifier;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Alias.java:38]: add back final
```
ImmutableList.copyOf(Objects.requireNonNull(qualifier, "qualifier can not be null"));
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 3]
```diff
@@ -398,6 +399,12 @@ public PlanFragment visitPhysicalNestedLoopJoin(PhysicalNestedLoopJoin<Plan, Pla
     @Override
     public PlanFragment visitPhysicalProject(PhysicalProject<Plan> project, PlanTranslatorContext context) {
         PlanFragment inputFragment = project.child(0).accept(this, context);
+
+        project.getProjects().stream().filter(p -> p instanceof Alias).forEach(p -> {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:403]:
```suggestion
        project.getProjects().stream().filter(Alias.class::isInstance).forEach(p -> {
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 4]
```diff
@@ -398,6 +399,12 @@ public PlanFragment visitPhysicalNestedLoopJoin(PhysicalNestedLoopJoin<Plan, Pla
...
+            SlotRef ref = context.findSlotRef(((NamedExpression) p.child(0)).getExprId());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:404]: add a todo yo handle Alias's child is not NamedExpression

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/batch/MergeConsecutiveProjectJob.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+     * @param ctx planner context for execute job
+     */
+    public MergeConsecutiveProjectJob(CascadesContext ctx) {
+        //TODO: eliminate consecutive projects for view
+        super(ctx);
+        rulesJob.addAll(ImmutableList.of(
+                bottomUpBatch(ImmutableList.of(
+                        new MergeConsecutiveProjects()
+                        )
+                )
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/batch/MergeConsecutiveProjectJob.java:41]:
```suggestion
                bottomUpBatch(ImmutableList.of(new MergeConsecutiveProjects()))
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSlotReference.java [hunk 6]
```diff
@@ -269,38 +269,40 @@ private BoundStar bindQualifiedStar(List<String> qualifierStar, Void context) {
...
-                                        + StringUtils.join(qualifier, "."));
-                        }
-                    default:
-                        throw new AnalysisException("Not supported name: "
-                            + StringUtils.join(nameParts, "."));
+                if (nameParts.size() == 1) {
+                    return nameParts.get(0).equalsIgnoreCase(boundSlot.getName());
+                } else if (nameParts.size() <= 3) {
+                    int size = nameParts.size();
+                    return handleNamePartsLessThanThree(boundSlot, nameParts.subList(size - 2, size));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSlotReference.java:276]: could merge nameParts.size() == 1 branch into it?

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/MergeConsecutiveProjects.java [hunk 7]
```diff
@@ -50,6 +52,40 @@
...
 
 public class MergeConsecutiveProjects extends OneRewriteRuleFactory {
+
+    private static class ExpressionReplacer extends DefaultExpressionRewriter<Map<Expression, Expression>> {
+        public static final ExpressionReplacer INSTANCE = new ExpressionReplacer();
+
+        /**
+         * case 1: project(project(alias, alias) -> alias, alias) => project(alias, alias)
+         * case 2: project(project(alias, alias) -> slot ref, slot ref) => project(project(alias, alias))
+         * case 3: others: use ExpressionReplacer.
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/MergeConsecutiveProjects.java:62]: this comment should add some concrete example.
like this
```java
case 1: 

project(alias(c) + 1, alias(c) + 1)                                  
       |                                                =>       project(slotRef(a) + 1 + 1, slotRef(b) + 1 + 1)
project(slotRef(a) + 1 as c, slotRef(b) + 1 as d)
```

**Review by 924060929 — APPROVED** [review]
LGTM

**Pull request merged**